backend/data/src/main/java/io/tolgee/security/JwtToken.java

```diff
@@ -1,31 +1,37 @@
 package io.tolgee.security;
 
 import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
 import io.jsonwebtoken.Jwts;
 
 import java.security.Key;
+import java.util.Date;
 
 
 public class JwtToken {
     private final String value;
-    private final Key key;
+
+    private final Jws<Claims> parsed;
 
     public JwtToken(String value, Key key) {
         this.value = value;
-        this.key = key;
+        this.parsed = Jwts.parserBuilder().setSigningKey(key).build().parseClaimsJws(value);
     }
```

Comment:
When I have expired token it throws error 500 now, since the token is validated here, which didn't happen before. Please add some test that expired token throws only 401.

Reply:
Maybe it would be better to not parse the key here at all. And let it for the io.tolgee.security.JwtTokenProviderImpl.validateToken